packages/contracts/test/core/dao/dao.ts

```diff
@@ -1198,90 +1230,174 @@ describe('DAO', function () {
       ).to.be.eq(true);
     });
 
-    it('should return 0 if no validator is set', async () => {
-      expect(
-        await dao.isValidSignature(ethers.utils.keccak256('0x00'), '0x00')
-      ).to.be.eq('0x00000000');
+    it('reverts all signatures requests by default', async () => {
+      const caller = signers[0];
+      const signer = signers[1];
+
+      const message = 'Hi!';
+      const hash = ethers.utils.hashMessage(message);
+      const signature = await signer.signMessage(message);
+
+      // The caller can validate signatures now.
+      expect(await dao.connect(caller).isValidSignature(hash, signature)).to.be
+        .reverted;
     });
 
-    it('should allow only `SET_SIGNATURE_VALIDATOR_PERMISSION_ID` to set validator', async () => {
-      await expect(
-        dao
-          .connect(signers[2])
-          .setSignatureValidator(ethers.Wallet.createRandom().address)
-      )
+    it('allows caller specific signature validation', async () => {
+      const caller = signers[0];
+      const signer = signers[1];
+
+      const message = 'Hi!';
+      const hash = ethers.utils.hashMessage(message);
+      const goodSignature = await signer.signMessage(message);
+      const badSignature = await signer.signMessage('Bye!');
+      expect(badSignature).to.not.equal(goodSignature);
+
+      // Try to call with caller but caller has no permission
+      expect(await dao.connect(caller).isValidSignature(hash, goodSignature))
         .to.be.revertedWithCustomError(dao, 'Unauthorized')
-        .withArgs(
-          dao.address,
-          signers[2].address,
-          PERMISSION_IDS.SET_SIGNATURE_VALIDATOR_PERMISSION_ID
-        );
-    });
+        .withArgs();
 
-    it('should set validator and emits event', async () => {
-      const validatorAddress = ethers.Wallet.createRandom().address;
-      const tx = await dao.setSignatureValidator(validatorAddress);
+      // Grant the permission to validate signatures to the caller unconditionally
+      await dao.grant(
+        dao.address,
+        caller.address,
+        PERMISSION_IDS.VALIDATE_SIGNATURE_PERMISSION_ID
+      );
 
-      expect(await dao.signatureValidator()).to.be.eq(validatorAddress);
+      // The caller can validate signatures now.
+      expect(await dao.connect(caller).isValidSignature(hash, goodSignature)).to
+        .not.be.reverted;
 
-      await expect(tx)
-        .to.emit(dao, EVENTS.SignatureValidatorSet)
-        .withArgs(validatorAddress);
+      // Because the caller is allowed unconditionally, the validation is always true.
+      expect(
+        await dao.connect(caller).isValidSignature(hash, goodSignature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
+      expect(
+        await dao.connect(caller).isValidSignature(hash, badSignature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
```

Comment:
I fully don't get what is the purpose behind having `badSignature`!
why it is bad?
if we must have 2 signatures, perhaps we can rename then to `currentSignature` & `futuresSignature` or something similar.

Reply:
It is a signature that doesn't match with the hash of the message "Hi!" that we are putting into `isValidSignature`.
In some tests for the bypassing feature, I use it to show that the actual signature and hash are irrelevant as long as the permission is granted (unconditionally) on the calling contract.

Reply:
Fixed in 25b7704e70b50f8b8a546560083e551d0aeab194